knn_demonstrations_test.ipynb

```diff
@@ -0,0 +1,323 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dsp\n",
+    "import os\n",
+    "import openai\n",
+    "# api_key = '<replace with your API key>'\n",
+    "# openai.api_key = api_key\n",
+    "\n",
+    "os.environ['KMP_DUPLICATE_LIB_OK'] = 'TRUE'"
```

Comment:
I don't recognize this. Is it necessary & safe? Will people need to use it everytime they use the kNN functions?

Reply:
No, they don't. The reason I added this is that I use VS Code, and it (seems) actually runs everything under the hood from the same kernel. Thus, I just can't use FAISS if I already imported/used it during the session. 

KMP_DUPLICATE_LIB_OK=True tells OpenMP to not complain if it notices that two copies of OpenMP are loaded. Don't see how this can hurt safety. If people use Jupyter - it would work well even without this env variable. But I expect some people will get the same error as I got, thus, I decided to keep this row in the demo ipynb.